lisp/ess-custom.el

```diff
@@ -2088,7 +2098,7 @@ See also function `ess-create-object-name-db'.")
 (defvar ess-R-function-name-regexp
   (concat "\\("      "\\sw+" "\\)"
           "[ \t]*"   "\\(<-\\)"
-          "[ \t\n]*" "function\\b"))
+          "[ \t\n]*" "\\(function\\b\\|\\\\\\)"))
```

Comment:
Personally I only use the shorthand notation for lambdas, and I agree with you but since the options has been made available by R Core, I think it is a good thing to have it supported by ESS. I see your point concerning the 'encouragement', but I do not consider this as a real one (including snippet templates would have been a different story^^)